app/page.tsx

```diff
@@ -18,10 +18,50 @@ export default function Home() {
   const [liveClasses, setLiveClasses] = useState<SessionOccurrence[]>([]);
   const [isLoading, setIsLoading] = useState(true);
   const [quizzes, setQuizzes] = useState<QuizSession[]>([]);
+  const [quizCompletionStatus, setQuizCompletionStatus] = useState<QuizCompletionStatus>({});
   const commonTextClass = "text-gray-700 text-xs md:text-sm mx-3 md:mx-8 whitespace-nowrap w-12";
   const infoMessageClass = "flex items-center justify-center text-center h-72 mx-4 pb-40";
   const portalBaseUrl = api.portal.frontend.baseUrl;
 
+  const fetchQuizCompletionStatus = async () => {
+    try {
+      const response = await fetch(`${process.env.NEXT_PUBLIC_PROD_QUIZ_BACKEND_URL}sessions/user/${userId}/quiz-attempts`);
+      const data = await response.json();
+      setQuizCompletionStatus(data);
+    } catch (error) {
+      console.error("Error fetching quiz completion status:", error);
+    }
+  };
+
+  const isQuizAttemptable = (platformId: string) => {
+    return !quizCompletionStatus.hasOwnProperty(platformId) ||
+      quizCompletionStatus[platformId] === false;
+  };
+
+  const filterAndSortTests = (quizzes: QuizSession[]) => {
```

Comment:
Add a comment here explaining what the method does and why it is needed on Gurukul. Other methods are obvious, so not needed